From 32ef7dc5316ddaa9d1357b0456b2809cf47524ee Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Tue, 2 Feb 2021 08:25:47 +0800
Subject: [PATCH] kernel shutdown prepare add ignore flag

---
 kernel/reboot.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kernel/reboot.c b/kernel/reboot.c
index 47a94a9a5..996ec1a61 100644
--- a/kernel/reboot.c
+++ b/kernel/reboot.c
@@ -266,6 +266,12 @@ EXPORT_SYMBOL_GPL(kernel_restart);
 
 static void kernel_shutdown_prepare(enum system_states state)
 {
+	pr_info("kernel shutdown prepare\n");
+	if ( reboot_kernel_restart_prepare_disabled ) {
+		pr_info("kernel shutdown prepare was ignored\n");
+		return;
+	}
+
 	blocking_notifier_call_chain(&reboot_notifier_list,
 		(state == SYSTEM_HALT) ? SYS_HALT : SYS_POWER_OFF, NULL);
 	system_state = state;
-- 
2.25.1

