From a102ee1ba897d22aa9d0557cd2907f25e3a8a7bc Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Fri, 26 Feb 2021 10:25:05 +0800
Subject: [PATCH] drm: meson_drv: add remove call

Signed-off-by: Artem Lapkin <art@khadas.com>

---
 drivers/gpu/drm/meson/meson_drv.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/meson/meson_drv.c b/drivers/gpu/drm/meson/meson_drv.c
index ac3808e84..44bd96027 100644
--- a/drivers/gpu/drm/meson/meson_drv.c
+++ b/drivers/gpu/drm/meson/meson_drv.c
@@ -493,6 +493,13 @@ static void meson_drv_shutdown(struct platform_device *pdev)
 	drm_atomic_helper_shutdown(drm);
 }
 
+static int meson_drv_remove(struct platform_device *pdev)
+{
+	dev_warn(&pdev->dev, "%s", __func__);
+	component_master_del(&pdev->dev, &meson_drv_master_ops);
+	return 0;
+}
+
 static int meson_drv_probe(struct platform_device *pdev)
 {
 	struct component_match *match = NULL;
@@ -565,6 +572,7 @@ static const struct dev_pm_ops meson_drv_pm_ops = {
 static struct platform_driver meson_drm_platform_driver = {
 	.probe      = meson_drv_probe,
 	.shutdown   = meson_drv_shutdown,
+	.remove     = meson_drv_remove,
 	.driver     = {
 		.name	= "meson-drm",
 		.of_match_table = dt_match,
-- 
2.25.1

